logger/umlogger_test.go

```diff
@@ -0,0 +1,278 @@
+// Package logger provides implementetion of writing Log messages to Graylog, file, and stdout.
+// Supports log levels and destination.
+package logger
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	loggerHost       = "loggerUM.Host"
+	loggerPort       = "loggerUM.Port"
+	loggerPassSecret = "loggerUM.PassSecret"
+	loggerPassSHA2   = "loggerUM.PassSHA2"
+	loggerOutput     = "loggerUM.Output"
+)
+
+func TestNullFormatter_Format(t *testing.T) {
+	type args struct {
+		e *log.Entry
+	}
+	formatter := &log.JSONFormatter{}
+	b, err := formatter.Format(log.WithField("opg", errors.New("user managment test")))
+	if err != nil {
+		t.Fatal("Unable to format entry: ", err)
+	}
+
+	entry1 := make(map[string]interface{})
+	err = json.Unmarshal(b, &entry1)
+	if err != nil {
+		t.Fatal("Unable to unmarshal formatted entry: ", err)
+	}
+	logger := log.New()
+	logger.Out = &bytes.Buffer{}
+	entry := log.NewEntry(logger)
+
+	tests := []struct {
+		name    string
+		n       log.Formatter
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{name: "Null format",
+			n:       NullFormatter{},
+			args:    args{e: entry},
+			want:    []byte{},
+			wantErr: false,
+		}, {
+			name:    "JSON format",
+			n:       NullFormatter{},
+			args:    args{e: &log.Entry{Logger: logger, Data: entry1}},
+			want:    []byte{},
+			wantErr: false,
+		},
```

Comment:
Want will never be anything but empty slice of bytes. You even don't need this param.

Reply:
And how I must to test this function?
t.Errorf("NullFormatter.Format() = %v, want %v", got,[]byte{})?

Reply:
fixed